static/app/utils/discover/genericDiscoverQuery.tsx

```diff
@@ -127,7 +130,7 @@ type ComponentProps<T, P> = {
   /**
    * Allows components to modify the payload before it is set.
    */
-  getRequestPayload?: (props: Props<T, P>) => any;
+  getRequestPayload?: (props: Omit<Props<T, P>, 'api'>) => any;
```

Comment:
Have to fight with these generics still since I'd prefer to not pass api anymore.